### DISCUSSION TITLE
Running vaultwarden service from binary: 404 on root, but admin working

### DISCUSSION THREAD

**Discussion by GRbit** [dani-garcia/vaultwarden#3187] [Q&A] [answered]
I've set up a systemd vaultwarden service on my little old orangePi machine, and it kind of works, but not as I expected.
I use a subdomain provided from duckdns.org to access my orangePi, and I can access the admin page and log in using a token, but I can't access the regular service. I didn't set up any proxies, it just works as is.

Here are some settings that are related to the problem:
```
DOMAIN=http://my-domain.duckdns.org:18080
ROCKET_ADDRESS=0.0.0.0
ROCKET_PORT=18080
```

Here are some logs, showing that it's ok to access an admin page, but "/" request returns a 404
```
[2023-01-28 15:44:32.005][request][INFO] GET /admin/diagnostics
[2023-01-28 15:44:33.460][response][INFO] (diagnostics) GET /admin/diagnostics => 200 OK
[2023-01-28 15:44:51.919][_][ERROR] No matching routes for GET / text/html.
[2023-01-28 15:44:51.919][_][WARN] No 404 catcher registered. Using Rocket default.
```

So, I don't know what am I doing wrong and what should I check. Please, if know what could be wrong, help me.

**Comment by stefan0xC** [comment, marked as answer]
You are probably missing the [web-vault](https://github.com/dani-garcia/vaultwarden/wiki/Building-binary#install-the-web-vault). You can tell vaultwarden where it is by setting `WEB_VAULT_FOLDER` (it defaults to `web-vault/` in the working directory of the process)

**Reply by GRbit** [reply]
> You are probably missing the [web-vault](https://github.com/dani-garcia/vaultwarden/wiki/Building-binary#install-the-web-vault). You can tell vaultwarden where it is by setting `WEB_VAULT_FOLDER` (it defaults to `web-vault/` in the working directory of the process)

I set `WEB_VAULT_ENABLE=false`. Is it mandatory to run vaultwarden?

**Reply by GRbit** [reply]
Ok, seems like it's mandatory. Thank you @stefan0xC !

I thought that since admin is accessible, then it should work fine. Do you think it makes sense to not just return 404, but something like "web-vault is needed for this functionality"? maybe it'll save some time to others in the future.
